src/libstore/store-api.hh

```diff
@@ -115,6 +115,7 @@ struct ValidPathInfo
 {
     StorePath path;
     std::optional<StorePath> deriver;
+    std::optional<string> outputname;
```

Comment:
It seems incorrect to have an `outputName` field in `ValidPathInfo`. A CA store path can be produced by differently named output of different derivations (e.g. the "foo" output of derivation A could produce the same contents as the "bar" output of derivation B).